platforms/core-runtime/native/src/main/java/org/gradle/internal/nativeintegration/jansi/JansiBootPathConfigurer.java

```diff
@@ -41,41 +43,28 @@ public class JansiBootPathConfigurer {
      */
     public void configure(File storageDir) {
         JansiStorage jansiStorage = locator.locate(storageDir);
-
         if (jansiStorage != null) {
             File libFile = jansiStorage.getTargetLibFile();
             libFile.getParentFile().mkdirs();
-
             if (!libFile.exists()) {
-                InputStream libraryInputStream = getClass().getResourceAsStream(jansiStorage.getJansiLibrary().getResourcePath());
-                try {
-                    if (libraryInputStream != null) {
-                        copyLibrary(libraryInputStream, libFile);
-                    }
-                } finally {
-                    IoActions.closeQuietly(libraryInputStream);
-                }
+                copyLibraryQuietly(
+                    getClass().getResourceAsStream(jansiStorage.getJansiLibrary().getResourcePath()),
+                    libFile
+                );
             }
-
-            System.setProperty(JANSI_LIBRARY_PATH_SYS_PROP, libFile.getParent());
+            setProperty(JANSI_LIBRARY_PATH_SYS_PROP, libFile.getParent());
         }
     }
 
-    private void copyLibrary(InputStream lib, File libFile) {
```

Comment:
imho it was doing copyLibraryQuietly already hiding the impl. detail on the only point where it matters, on the top abstraction layer.